bw_add_sshkeys.py

```diff
@@ -220,36 +164,25 @@ def ssh_add(session, item_id, key_id):
     logging.debug('Item ID: %s', item_id)
     logging.debug('Key ID: %s', key_id)
 
-    # TODO: avoid temporary files, if possible
-    with tempfile.NamedTemporaryFile() as tmpfile:
-        proc = subprocess.Popen(
-            list(filter(None, [
-                'bw',
-                (None, '--nointeraction')[cli_supports('nointeraction')],
-                '--quiet',
-                'get',
-                'attachment', key_id,
-                '--itemid', item_id,
-                '--output', tmpfile.name,
-                '--session', session
-            ])),
-            stderr=subprocess.PIPE
-        )
-        (_, stderr) = proc.communicate()
-        if proc.returncode:
-            logging.error(stderr.decode('utf-8'))
-            return False
-
-        logging.debug("Running ssh-add")
-
-        # CAVEAT: `ssh-add` provides no useful output, even with maximum verbosity
-        proc = subprocess.Popen(['ssh-add', tmpfile.name])
-        proc.wait()
+    proc_attachment = subprocess.run([
+            'bw',
+            'get',
+            'attachment', key_id,
+            '--itemid', item_id,
+            '--raw',
+            '--session', session
+        ],
+        stdout=subprocess.PIPE,
+        text=True,
+    )
+    proc_attachment.check_returncode()
+    ssh_key = proc_attachment.stdout
 
-        if proc.returncode:
-            return False
+    logging.debug("Running ssh-add")
 
-        return True
+    # CAVEAT: `ssh-add` provides no useful output, even with maximum verbosity
+    proc_ssh_add = subprocess.run(['ssh-add', '-'], input=ssh_key, text=True)
+    proc_ssh_add.check_returncode()
```

Comment:
With these changes, the SSH keys are added, but an error message is being printed.
Need to return True here, or change the ssh_add call on line 156.